### PR TITLE
ACMS-858: Move content tests back to its module

### PR DESCRIPTION
**Pull request by chandan-singh7929**
**Motivation**

Fixes #[ACMS-858](https://backlog.acquia.com/browse/ACMS-858)

**Proposed changes**

This fixes test for all components modules

**Alternatives considered**

NA
**Testing steps**

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

File: modules/acquia_cms_article/tests/src/ExistingSite/ArticleListTest.php [hunk 1]
```diff
@@ -25,8 +25,11 @@ class ArticleListTest extends ContentTypeListTestBase {
   /**
    * {@inheritdoc}
    */
-  protected function getView() : View {
-    return View::load('articles');
+  protected function getView() {
+    if ($this->container->get('module_handler')->moduleExists('acquia_cms_search')) {
+      return View::load('articles');
```
Comment (kdruckman) [inline modules/acquia_cms_article/tests/src/ExistingSite/ArticleListTest.php:30]: I think for the scope of this PR, this is ok, but I'm concerned that we don't have an articles view in the absence of the search module

File: modules/acquia_cms_common/tests/src/ExistingSite/ContentTypeListTestBase.php [hunk 2]
```diff
@@ -208,79 +207,86 @@ public function testListPage(array $permissions = NULL) {
...
-    $this->drupalGet('/search', $options);
-    $assert_session->linkExists('Alpha');
-    $assert_session->linkNotExists('Spanish - Alpha');
-
-    $this->drupalGet('/es/search', $options);
-    $assert_session->linkExists('Spanish - Alpha');
+    // Since below tests would require certain module to be present
+    // in order to pass all test, we are adding check instead of
+    // adding the required module as dependencies for excellent
+    // reasons so that we can run the minimum tests in isolation.
```
Comment (kdruckman) [inline modules/acquia_cms_common/tests/src/ExistingSite/ContentTypeListTestBase.php:213]: I'd remove "for excellent reasons" here.

**Review by kdruckman — CHANGES_REQUESTED** [review]
I think this is looking pretty good, but I think we should have a separate conversation about the article view in the absence of search.

File: modules/acquia_cms_common/tests/src/ExistingSite/ContentTypeListTestBase.php (hunk 2 above)
Comment (chandan-singh7929) [reply modules/acquia_cms_common/tests/src/ExistingSite/ContentTypeListTestBase.php]: Done.

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**